sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/DisposableDatasourceCredential.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.AI.MetricsAdvisor.Administration;
+using Azure.AI.MetricsAdvisor.Models;
+using NUnit.Framework;
+
+namespace Azure.AI.MetricsAdvisor.Tests
+{
+    /// <summary>
+    /// Represents an <see cref="DatasourceCredential"/> that has been created for testing purposes.
+    /// In order to create a new instance of this class, the <see cref="CreateDatasourceCredentialAsync"/>
+    /// static method must be invoked. The created datasource credential will be deleted upon disposal.
+    /// </summary>
+    public class DisposableDatasourceCredential : IAsyncDisposable
+    {
+        /// <summary>
+        /// The client to use for deleting the datasource credential upon disposal.
+        /// </summary>
+        private readonly MetricsAdvisorAdministrationClient _adminClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DisposableDatasourceCredential"/> class.
+        /// </summary>
+        /// <param name="adminClient">The client to use for deleting the credential entity upon disposal.</param>
+        /// <param name="credential">The datasource credential this instance is associated with.</param>
+        private DisposableDatasourceCredential(MetricsAdvisorAdministrationClient adminClient, DatasourceCredential credential)
+        {
+            _adminClient = adminClient;
+            Credential = credential;
+        }
+
+        /// <summary>
+        /// The datasource credential this instance is associated with.
+        /// </summary>
+        public DatasourceCredential Credential { get; }
```

Comment:
We're not storing the ID anymore, but the whole credential. When we started using these "Disposable" types, we were only returning the ID of the created entities to the user. Now we do an internal GET operation to return the whole model, so we're already exposing it here.

I still need to update other "disposable" types to do the same, but definitely not a priority right now.

Reply:
Now I do (https://github.com/Azure/azure-sdk-for-net/issues/21583).